Client/core/CDiscordRichPresence.cpp

```diff
@@ -0,0 +1,222 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        core/CDiscordRichPresence.cpp
+ *  PURPOSE:     Discord rich presence implementation
+ *
+ *  Multi Theft Auto is available from http://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "discord_rpc.h"
+#include "CDiscordRichPresence.h"
+
+constexpr char DEFAULT_APP_ID[] = "1139666477813866546";
+constexpr char DEFAULT_APP_ASSET[] = "mta_logo_round";
+constexpr char DEFAULT_APP_ASSET_TEXT[] = "Multi Theft Auto";
+constexpr char DEFAULT_APP_ASSET_SMALL[] = "";
+constexpr char DEFAULT_APP_ASSET_SMALL_TEXT[] = "";
```

Comment:
Nice work and PR, this feature has been awaited for a long time! :heart:

As far as I can see, scripts can customize rich presence information without changing the app ID, which in the eyes of Discord would establish a link between MTA and whatever server owners decide to use rich presence for. Should we force scripts to use their own app ID if they want to change the default rich presence data?

Reply:
I aimed to enable the scripts to establish their own application, hence create their image (emblem, logo?) in Rich Presence and other essential information.

Reply:
Right. It is my understanding that application IDs should not be treated as private information, and any third party can try to extract a rich presence application ID and use it for a completely unrelated application, so these concerns probably don't mean much from a security standpoint. But I still find them interesting.

Reply:
> Should we force scripts to use their own app ID if they want to change the default rich presence data?

Good thinking. We should stop customization unless the server is using their own application ID.

> Also, out of curiosity, who manages the default app ID?

I think that's @znvjder's app ID. We'll get one created under the MTA discord, nice spot.

Reply:
Please address the first comment above - unless the server has set their own application ID they shouldn't be able to customize anything.

Reply:
The given ID in the default application was for the time of testing, sorry for that, so MTA team has to give their own ID.

Generally there is a variable to lock it, just need to add some conditions to the functions. But in general - do we need to block all features if we don't want to use a custom application?

Reply:
Yeah I'd say block everything

Reply:
> The given ID in the default application was for the time of testing, sorry for that, so MTA team has to give their own ID.
> 
> Generally there is a variable to lock it, just need to add some conditions to the functions. But in general - do we need to block all features if we don't want to use a custom application?

I think every server should set-up their own application id.

Reply:
Already all features are blocked till we set up own appliaction. :)